### ISSUE TITLE
Some Mandatory To Implement Stats fields are out of sync with webrtc-stats

### ISSUE THREAD

**Issue by soareschen** [w3c/webrtc-pc#1474]
In section 8.6, there are a number of fields that are either renamed or removed from the latest webrtc-stats. For example in `RTCRTPStreamStats` the fields `associateStatsId`,  `isRemote`, `mediaTrackId` are not found.

There are also some fields that are indicated as necessarily optional in webrtc-stats, but required in section 8.6. For example the `issuerCertificateId` in `RTCCertificateStats` is stated to be not set _"If the current certificate is at the end of the chain"_.

**Comment by aboba** [comment]
@soareschen The WebRTC-stats document is clear that WebRTC-PC Section 8.6 is definitive:

"Specifications that refer to this specification have the need to specify conformance. They should put in their document text like this:"

**Comment by soareschen** [comment]
Does this means issues have to be filed on webrtc-stats to define the missing fields? If the fields are not defined in webrtc-stats, they cannot be made mandatory in webrtc-pc alone.

**Comment by aboba** [comment]
@jan-ivar @vr00m Can you take a look at this?

**Comment by soareschen** [comment]
I have compiled here a list of mandatory fields mentioned in webrtc-pc but not found in webrtc-stats:

- RTCCodecStats - `codec` (should be `codecType`?), `parameters`
- RTCOutboundRTPStreamStats - `roundTripTime`
- RTCMediaStreamTrackStats - `ssrcIds`
- RTCDataChannelStats - `dataChannelId` (should be `datachannelid`?)
- RTCTransportStats - `activeConnection`
- RTCIceCandidatePairStats - `writable`, `readable`, `totalRtt` (should be `totalRoundTripTime`?), `currentRtt` (should be `currentRoundTripTime`?)

**Comment by jan-ivar** [comment]
TL;DR: Have the conformance criteria thus far been basically "blanket implement all members"? If so, a blanket update of webrtc-pc to match webrtc-stats should suffice.

Renaming is simple, but how to handle new fields? We probably should require them, unless there were fields explicitly left out last time the conformance criteria were updated. Anyone know? 

Thanks @soareschen for the list! Stats innovation generally happens in webrtc-stats, so webrtc-pc is lagging, and should be updated, in spite of the official conformance criteria pointing the other way. 

* RTCCodecStats - See https://github.com/w3c/webrtc-stats/pull/168 (which ironically came from webrtc-pc).
* RTCRTPStreamStats - s/`associateStatsId`/`remoteId`/ + `localId`, s/`isRemote`//, s/`mediaTrackId`/`trackId`/ - from https://github.com/w3c/webrtc-stats/pull/191.
* `roundtripTime` moved to [RTCRemoteInboundRTPStreamStats](https://w3c.github.io/webrtc-stats/#dom-rtcremoteinboundrtpstreamstats) - also https://github.com/w3c/webrtc-stats/pull/191.

Those are the ones from your list I know off the top of my head.

**Comment by jan-ivar** [comment]
I'm happy to submit a PR for the ones that can be pinned on me at least.

**Comment by aboba** [comment]
@jan-ivar A PR would be welcome, thanks!

**Comment by jan-ivar** [comment]
@aboba I added https://github.com/w3c/webrtc-pc/issues/1480 already for rtp. PTAL 😏

I'll look at the rest on Monday if @vr00m doesn't beat me to it.
